src/pages/workspaces/workspace/workflows/DataStepContent.js

```diff
@@ -41,10 +43,10 @@ export default class DataStepContent extends Component {
   isValidSelectionModel() {
     const { entitySelectionModel } = this.state
     const { newSetName, selectedEntities, type } = entitySelectionModel
-    const { entityType, name } = selectedEntities
-    return (type === EntitySelectionType.processAll ||
-      (type === EntitySelectionType.processFromSet && !!entityType && !!name) ||
-      (_.size(selectedEntities) > 0 && !!newSetName))
+    const selectionSize = _.size(selectedEntities)
+
+    return selectionSize === 1 ||
+      (!!newSetName && (type === processAll || (selectionSize > 1 && (type === chooseRows || selectionSize <= 10))))
```

Comment:
nit: technically don't need to validate `newSetName` in the `chooseSet` case, but since it gets initialized to a value it doesn't seem to cause problems.

Reply:
➕ More than a nit to me. I want to make sure we leave this code in a state where it is obviously doing the right thing. If it looks like there's a case that isn't fully considered or is making needless checks, then we won't know what the correct behavior is supposed to me when we come back to this code a year from now... and the corner cases are too complex for us to reliably always test all of them manually when making changes.